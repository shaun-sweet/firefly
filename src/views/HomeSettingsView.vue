<template>
  <div class="layout-padding">
    <q-list no-border>
      <q-list-header>
        <h5>Owners</h5>
      </q-list-header>
      <q-item>
        <q-item-main class="owners" link dense multiline>
          <div class="owners-information">
            <div class="owner-name">Display Name: <span>{Shaun}</span></div>
            <div class="owner-email">Email: <span>{email@email.com}</span></div>
          </div>
        </q-item-main>
      </q-item>

      <q-item-separator />
    </q-list>

    <q-list no-border>
      <q-list-header>
        <h5>Guests</h5>
      </q-list-header>

      <q-item>
        <q-item-main class="guests" link dense multiline>
          <div class="guests-information">
            <div class="guests-name">Display Name: <span>{Shaun}</span></div>
            <div class="guests-email">Email: <span>{email@email.com}</span></div>
            <div class="guests-home-nickname">Home Nickname: <span>{Zach Priddys Firefly Home}</span></div>
          </div>
        </q-item-main>
        <q-item-side right>
          <div class="role-selector">
            <q-field>
              <q-select
              separator
              stack-label="Role"
              v-model="select"
              :options="selectOptions"
              />
            </q-field>
          </div>
        </q-item-side>
        <q-item-side right icon="ion-ios-trash-outline" />
      </q-item>

      <q-item-separator class="separator" />

      <q-item>
        <q-item-main class="guests" link dense multiline>
          <div class="guests-information">
            <div class="guests-name">Display Name: <span>{Amy}</span></div>
            <div class="guests-email">Email: <span>{email@email.com}</span></div>
            <div class="guests-home-nickname">Home Nickname: <span>{Zach Priddys Firefly Home}</span></div>
          </div>
        </q-item-main>
        <q-item-side right>
          <div class="role-selector">
            <q-field>
              <q-select
              separator
              float-label="Role"
              v-model="select"
              :options="selectOptions"
              />
            </q-field>
          </div>
        </q-item-side>
        <q-item-side right icon="ion-ios-trash-outline" />
      </q-item>
    </q-list>

    <q-list no-border>
      <q-list-header id="invite-to-home">
          <h5>Pending Invite(s)</h5>
          <q-btn small @click="$refs.inviteUserMenu.open()" outline class="button" color="primary" icon="ion-person-add"> Invite a Friend!</q-btn>
      </q-list-header>

      <q-item>
        <q-item-main class="owners" link dense multiline>
          <div class="owners-information">
            <div class="owner-name">Display Name: <span>{Shaun}</span></div>
            <div class="owner-email">Email: <span>{email@email.com}</span></div>
          </div>
        </q-item-main>
      </q-item>
      <q-item-separator class="separator" />

    </q-list>

    <q-modal maximized ref="inviteUserMenu">
      <q-modal-layout>
        <q-toolbar slot="header">
          <div class="q-toolbar-title">
            Invite a friend
          </div>
        </q-toolbar>
        <q-toolbar slot="footer">
          <div class="q-toolbar-title">
            Firefly Home Automation
          </div>
        </q-toolbar>
        <div class="layout-padding">
          <q-field>
            <q-input
              type="text"
              stack-label="Name"
              v-model="email" />
          </q-field>
          <q-field>
            <q-input
              type="email"
              stack-label="Email Address"
              v-model="email" />
          </q-field>
          <q-field>
            <q-input
              type="text"
              stack-label="Home Nickname"
              v-model="email" />
          </q-field>
          <q-select
          separator
          stack-label="Role"
          v-model="select"
          :options="selectOptions"
          />
          <q-btn color="primary" @click="$refs.inviteUserMenu.close()">Submit</q-btn>
          <q-btn outline @click="$refs.inviteUserMenu.close()">Close</q-btn>
        </div>
      </q-modal-layout>
    </q-modal>

  </div>
</template>

<script>
export default {
  data () {
    return {
      email: '',
      select: 'guest',
      selectOptions: [
        {
          label: 'Owner',
          value: 'owner'
        },
        {
          label: 'Guest',
          value: 'guest'
        }
      ]
    }
  }
}
</script>

<style lang="stylus" scoped>
#invite-to-home
  display flex
  .button
    margin-left 2em

.owners-information > div, .guests-information > div
  margin-bottom: .5em
  padding-left: 1em

.separator
  margin-left: 1em
</style>
