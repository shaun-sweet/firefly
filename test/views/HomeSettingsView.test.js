import Vue from 'vue';
import { shallow } from 'vue-test-utils';
import HomeSettingsView from 'src/views/HomeSettingsView.vue';

describe('HomeSettingsView', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(HomeSettingsView);
  })
  it('should render the view correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

})
