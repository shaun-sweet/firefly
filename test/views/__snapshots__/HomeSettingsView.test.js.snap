// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeSettingsView should render the view correctly 1`] = `
<div
  class="layout-padding"
>
  <q-list
    no-border=""
  >
    <q-list-header>
      <h5>
        Owners
      </h5>
    </q-list-header>
     
    <q-item>
      <q-item-main
        class="owners"
        dense=""
        link=""
        multiline=""
      >
        <div
          class="owners-information"
        >
          <div
            class="owner-name"
          >
            Display Name: 
            <span>
              {Shaun}
            </span>
          </div>
           
          <div
            class="owner-email"
          >
            Email: 
            <span>
              {email@email.com}
            </span>
          </div>
        </div>
      </q-item-main>
    </q-item>
     
    <q-item-separator />
  </q-list>
   
  <q-list
    no-border=""
  >
    <q-list-header>
      <h5>
        Guests
      </h5>
    </q-list-header>
     
    <q-item>
      <q-item-main
        class="guests"
        dense=""
        link=""
        multiline=""
      >
        <div
          class="guests-information"
        >
          <div
            class="guests-name"
          >
            Display Name: 
            <span>
              {Shaun}
            </span>
          </div>
           
          <div
            class="guests-email"
          >
            Email: 
            <span>
              {email@email.com}
            </span>
          </div>
           
          <div
            class="guests-home-nickname"
          >
            Home Nickname: 
            <span>
              {Zach Priddys Firefly Home}
            </span>
          </div>
        </div>
      </q-item-main>
       
      <q-item-side
        right=""
      >
        <div
          class="role-selector"
        >
          <q-field>
            <q-select
              options="[object Object],[object Object]"
              separator=""
              stack-label="Role"
            />
          </q-field>
        </div>
      </q-item-side>
       
      <q-item-side
        icon="ion-ios-trash-outline"
        right=""
      />
    </q-item>
     
    <q-item-separator
      class="separator"
    />
     
    <q-item>
      <q-item-main
        class="guests"
        dense=""
        link=""
        multiline=""
      >
        <div
          class="guests-information"
        >
          <div
            class="guests-name"
          >
            Display Name: 
            <span>
              {Amy}
            </span>
          </div>
           
          <div
            class="guests-email"
          >
            Email: 
            <span>
              {email@email.com}
            </span>
          </div>
           
          <div
            class="guests-home-nickname"
          >
            Home Nickname: 
            <span>
              {Zach Priddys Firefly Home}
            </span>
          </div>
        </div>
      </q-item-main>
       
      <q-item-side
        right=""
      >
        <div
          class="role-selector"
        >
          <q-field>
            <q-select
              float-label="Role"
              options="[object Object],[object Object]"
              separator=""
            />
          </q-field>
        </div>
      </q-item-side>
       
      <q-item-side
        icon="ion-ios-trash-outline"
        right=""
      />
    </q-item>
  </q-list>
   
  <q-list
    no-border=""
  >
    <q-list-header
      id="invite-to-home"
    >
      <h5>
        Pending Invite(s)
      </h5>
       
      <q-btn
        class="button"
        color="primary"
        icon="ion-person-add"
        outline=""
        small=""
      >
         Invite a Friend!
      </q-btn>
    </q-list-header>
     
    <q-item>
      <q-item-main
        class="owners"
        dense=""
        link=""
        multiline=""
      >
        <div
          class="owners-information"
        >
          <div
            class="owner-name"
          >
            Display Name: 
            <span>
              {Shaun}
            </span>
          </div>
           
          <div
            class="owner-email"
          >
            Email: 
            <span>
              {email@email.com}
            </span>
          </div>
        </div>
      </q-item-main>
    </q-item>
     
    <q-item-separator
      class="separator"
    />
  </q-list>
   
  <q-modal
    maximized=""
  >
    <q-modal-layout>
      <q-toolbar>
        <div
          class="q-toolbar-title"
        >
          
          Invite a friend
        
        </div>
      </q-toolbar>
       
      <q-toolbar>
        <div
          class="q-toolbar-title"
        >
          
          Firefly Home Automation
        
        </div>
      </q-toolbar>
       
      <div
        class="layout-padding"
      >
        <q-field>
          <q-input
            stack-label="Name"
            type="text"
          />
        </q-field>
         
        <q-field>
          <q-input
            stack-label="Email Address"
            type="email"
          />
        </q-field>
         
        <q-field>
          <q-input
            stack-label="Home Nickname"
            type="text"
          />
        </q-field>
         
        <q-select
          options="[object Object],[object Object]"
          separator=""
          stack-label="Role"
        />
         
        <q-btn
          color="primary"
        >
          Submit
        </q-btn>
         
        <q-btn
          outline=""
        >
          Close
        </q-btn>
      </div>
    </q-modal-layout>
  </q-modal>
</div>
`;
